pkg/kube/informer.go

```diff
@@ -0,0 +1,295 @@
+package kube
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+	ForResource(resource schema.GroupVersionResource) (InformerLister, error)
+	WaitForCacheSync(stopCh <-chan struct{})
+}
+
+type Informer interface {
+	AddEventHandler(handler cache.ResourceEventHandler)
+	AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration)
+	AddIndexers(indexers cache.Indexers) error
+	HasSynced() bool
+}
+
+type Lister interface {
```

Comment:
Why not implement `GenericLister`? Aside from being consistent/reusable I think the API is more friendly

Reply:
Nevermind I missed your comment. I guess both options are not great